src/navigation/MainNavigator.tsx

```diff
@@ -14,6 +15,11 @@ export const MainNavigator = () => {
           component={Connections}
           options={{headerShown: false}}
         />
+        <Stack.Screen
+          name={'BookingScreen'}
+          component={() => <Booking />}
```

Comment:
Pass the name of the component  like we did for connections (on line 15) instead of using a function. Passing a function will cause performance issues and possibly a problem with losing the component state on re-render.